test/test_common/environment.cc

```diff
@@ -65,12 +68,17 @@ bool TestEnvironment::shouldRunTestForIpVersion(const Network::Address::IpVersio
 
 std::vector<Network::Address::IpVersion> TestEnvironment::getIpVersionsForTest() {
   std::vector<Network::Address::IpVersion> parameters;
-  if (TestEnvironment::shouldRunTestForIpVersion(Network::Address::IpVersion::v4)) {
-    parameters.push_back(Network::Address::IpVersion::v4);
-  }
-
-  if (TestEnvironment::shouldRunTestForIpVersion(Network::Address::IpVersion::v6)) {
-    parameters.push_back(Network::Address::IpVersion::v6);
+  for (auto version : {Network::Address::IpVersion::v4, Network::Address::IpVersion::v6}) {
+    if (TestEnvironment::shouldRunTestForIpVersion(version)) {
+      parameters.push_back(version);
+      if (Network::Test::supportsIpVersion(version) == false) {
+        Logger::Registry::getLog(Logger::Id::testing)
+            .warn(fmt::format("Parameterized testing with IP{} addresses may not be supported."
+                              " If testing fails, set enviroment variable ENVOY_IP_TEST_VERSIONS "
+                              "in test/main.cc",
```

Comment:
You can drop the `main.cc` reference - most users will set `ENVOY_IP_TEST_VERSIONS` in the environment. Can you add some documentation to `bazel/README.md` to this effect? I think it will look something like:

```
bazel test //test/... --test_env=ENVOY_IP_TEST_VERSIONS=whatever
```